src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -140,7 +146,25 @@ bool ADU_are_correlation_ids_matching(
 
     if (outCorrelationData != NULL)
     {
-        if (ADUC_AllocAndStrCopyN(outCorrelationData, responseCorrelationId, (size_t)responseCorrelationIdLen) != 0)
+        // TODO: Remove this workaround once adu service deploys fix that ensures exact correlation id string agent sent is
+        // in the response message. Currently, it parses the UUID without dashes and then introduces dashes in the response
+        // message.
+        char correlation_id_no_dash[37] = "";
```

Comment:
don't init out params.

presumably 37 is GUID_LEN + 1?  If so, use a define instead of a magic number.  If it's coincidental that it's GUID_LEN, then create a "CORRELATION_ID_LEN" define.